PropertyEditDialog minor issues

le-sigwal opened the pull request:
Minor layout modification:
- Increased QDialog & QFormLayout height.
- Deleted unused items in QLineEdit

Initialization correction:
- Modified VibesGraphicsItem::parseJson() to initialize or correct json parameters EdgeColor/FaceColor/LineStyle/LineWidth of a VibesGraphicsGroup before being used in PropertyEditDialog
- Added VibesDefaults::replaceColorName() to replace short color name

godardma commented:
Thank you for your Pull Request ! I have a few remarks / questions about the suggested modifications : 


> Deleted unused items in QLineEdit
Well seen ! I didn't notice that this lines were unused.


> Increased QDialog & QFormLayout heigh
It is a good idea to give more space to the editor layout. However I find that there is a bit too much blank space between the last input line (Line Width) and the buttons. May I suggest something like modifying the height of line 10 to 190 and the one of line 20 to 140 ? It would give a result like : 

![Screenshot from 2025-05-05 16-00-20](https://github.com/user-attachments/assets/40627c7d-c695-4ec3-8188-06af6f7993bd)


> Modified VibesGraphicsItem::parseJson() to initialize or correct json parameters EdgeColor/FaceColor/LineStyle/LineWidth of a VibesGraphicsGroup before being used in PropertyEditDialog

I'm not really fan of this modification, here is a little example that i runned to test it : 
`#include "vibes.h"

int main()
{
    vibes::newFigure("Group figure");
    vibes::setFigureProperties(vibesParams("x",300,"y",40,"width",150,"height",150));

    vibes::newGroup("Group of boxes");

    vibes::drawBox(0.0,1.0,1.0,2.0,vibesParams("group","Group of boxes"));
    vibes::drawBox(1.0,2.0,1.0,2.0,vibesParams("group","Group of boxes","EdgeColor","green","FaceColor","yellow","LineStyle","-..","LineWidth","0.1"));
    return 0;
}`

In this example the group "Group of boxes" has no style. The example displays something like :
![Screenshot from 2025-05-05 16-14-02](https://github.com/user-attachments/assets/3f3dea0b-b4c6-450b-a4e3-9ca76277ef86)

Now if for example I use the editor to change the Face Color to blue, all the other properties are also forced on the objects, we then get : 
 
![Screenshot from 2025-05-05 16-12-55](https://github.com/user-attachments/assets/7a39d38a-3f4e-46a6-b8ba-403dfd845935)

While with the old implementation the properties of the group were uninitialized to say that they were item specific. Modifying only the Face Color then gave : 
 
![Screenshot from 2025-05-05 16-16-45](https://github.com/user-attachments/assets/e594fd0b-437d-41e7-8af4-d4b1b7e96dfd)


> Added VibesDefaults::replaceColorName() to replace short color name
From what I remember both short and long color names (say "r" and "red") are handled by Qt, did you encounter a case were the short names are an issue ?

le-sigwal commented:
Just added a commit for the PropertyEditDialog layout (190, 140, 121). 
godardma, could you tell if the rendering on your linux platform is closer to the following:
![vbs_vr_000001_taille-fenetre-corrigee_godardma](https://github.com/user-attachments/assets/a2ffe30d-6ccc-4eed-96d9-627ac48f7d0f)
On my windows platform, the rendering is tight vertically (but it may be a different issue):
![vbs_vr_000001_taille-fenetre-corrigee_gwin64](https://github.com/user-attachments/assets/9d52a9f2-7d80-47db-8598-b2c4550ed4ab)

godardma commented:
It is indeed really close from the picture you sent : 

![image](https://github.com/user-attachments/assets/ea25ff78-691f-4c24-8fb0-dbe875738242)

le-sigwal commented:
Just added a commit.
I didn't have the use-case of the "uninitialized group parameters" in mind -> so commit-back.
At the moment, I keep the function VibesDefaults::replaceColorName() that I have renamed VibesDefaults::toPredefinedColorName().

Let me explain.
The PropertyEditDialog uses Predifined Qt Color Name in the color Fields (Face and Edge).
When short name (or hexadecimal name) corresponding to a predefined color is specified, it is not displayed in PropertyEditDialog (because they are not in the list)
The function VibesDefaults::toPredefinedColorName() just replaces Hexadecimal and short color name to the corresponding Qt Predefined color name, as the following :
![toQtPredefinedColorName](https://github.com/user-attachments/assets/e277e9c0-0f99-4723-aed8-9d0cdb32e91c)
